Error when using projectdiscovery/chaos-client: panic: runtime error: invalid memory address or nil pointer dereference

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/chaos-client/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #chaos-client channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### chaos-client version:
v0.4.0

### Current Behavior:
I got some errors and scan whole terminal freezed while scanning for subdomains

### Command i used
chaos -d test.com | httpx -o live.txt

### Error i got 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x6b4028]

goroutine 6 [running]:
github.com/projectdiscovery/chaos-client/pkg/chaos.(*Client).do(0xc00007a240, 0xc0000f05d0)
        /home/sebin/go/pkg/mod/github.com/projectdiscovery/chaos-client@v0.4.0/pkg/chaos/chaos.go:37 +0x148
github.com/projectdiscovery/chaos-client/pkg/chaos.(*Client).GetSubdomains.func1(0xc000090120)
        /home/sebin/go/pkg/mod/github.com/projectdiscovery/chaos-client@v0.4.0/pkg/chaos/chaos.go:120 +0x1bb
created by github.com/projectdiscovery/chaos-client/pkg/chaos.(*Client).GetSubdomains
        /home/sebin/go/pkg/mod/github.com/projectdiscovery/chaos-client@v0.4.0/pkg/chaos/chaos.go:111 +0xe5
 
Screenshot: 
![image](https://user-images.githubusercontent.com/56429311/222668372-f6c94b47-8461-4774-b27d-f4849fe7804f.png)

--- a/pkg/chaos/chaos.go
+++ b/pkg/chaos/chaos.go
@@ -32,14 +32,12 @@ func (c *Client) do(request *retryablehttp.Request) (*http.Response, error) {
 		c.ratelimit.Take()
 	}
 	resp, err := c.httpClient.Do(request)
-	if err != nil {
-		if c.ratelimit == nil {
-			rl := resp.Header.Get("X-Ratelimit-Limit")
-			rlMax, er := strconv.Atoi(rl)
-			if er == nil {
-				// if er then ratelimit header is not present. Hence no rate limit
-				c.ratelimit = ratelimit.New(context.Background(), uint(rlMax), time.Minute)
-			}
+	if resp != nil && c.ratelimit == nil {
+		rl := resp.Header.Get("X-Ratelimit-Limit")
+		rlMax, err := strconv.Atoi(rl)
+		if err == nil && rlMax > 0 {
+			// if er then ratelimit header is not present. Hence no rate limit
+			c.ratelimit = ratelimit.New(context.Background(), uint(rlMax), time.Minute)
 		}
 	}
 	return resp, err

